pytorch_lightning/core/lightning.py

```diff
@@ -168,9 +168,10 @@ def example_input_array(self, example: Any) -> None:
 
     @property
     def datamodule(self) -> Any:
-        rank_zero_deprecation(
+        warning_cache.deprecation(
             "The `LightningModule.datamodule` property is deprecated in v1.3 and will be removed in v1.5."
-            " Access the datamodule through using `self.trainer.datamodule` instead."
+            " Access the datamodule through using `self.trainer.datamodule` instead.",
+            stacklevel=5,
```

Comment:
ugh.. figuring out the stacklevel is annoying isn't it? xD

Reply:
The changes aren't in this PR, but in #8002 you can just +1 the stacklevel in the function you are calling